### PR TITLE
Part of the implementation of registration update and unregister

### PR DESCRIPTION
**Pull request by peterwillcn**

### CODE REVIEW

File: blockchain/txvalidator.go [hunk 1]
```diff
@@ -1271,6 +1284,157 @@ func (b *BlockChain) checkRegisterCRTransaction(txn *Transaction) error {
 	return nil
 }
 
+func (b *BlockChain) getDIDAddress(info *payload.CRInfo) (string, error) {
```
Comment (raindust) [inline blockchain/txvalidator.go]: nice extraction and you can remove the same logic from checkRegisterCRTransaction method

Comment (raindust) [reply blockchain/txvalidator.go]: Ok, already removed, my mistake.

File: blockchain/txvalidator.go [hunk 2]
```diff
@@ -1214,60 +1305,84 @@ func (b *BlockChain) checkRegisterCRTransaction(txn *Transaction) error {
 		// check code and signature
 		if err := checkStandardSignature(program.Program{
 			Code:      info.Code,
-			Parameter: parameter,
+			Parameter: info.Signature,
```
Comment (RainFallsSilent) [inline blockchain/txvalidator.go]: should use:
	buf := new(bytes.Buffer)
	buf.WriteByte(byte(len(info.Signature)))
	buf.Write(info.Signature)
	parameter := buf.Bytes()

File: blockchain/txvalidator.go [hunk 3]
```diff
@@ -1214,60 +1305,84 @@ func (b *BlockChain) checkRegisterCRTransaction(txn *Transaction) error {
...
-		}
-		if !info.DID.IsEqual(*hash) {
-			return errors.New("the DID needs to be calculated from standard code")
-		}
 	} else if signType == vm.CHECKMULTISIG {
 		// check code and signature
 		if err := checkMultiSigSignatures(program.Program{
 			Code:      info.Code,
-			Parameter: parameter,
+			Parameter: info.Signature,
```
Comment (RainFallsSilent) [inline blockchain/txvalidator.go]: should use:
buf := new(bytes.Buffer)
buf.WriteByte(byte(len(info.Signature)))
buf.Write(info.Signature)
parameter := buf.Bytes()

File: blockchain/txvalidator.go (hunk 1 above)
Comment (peterwillcn) [reply blockchain/txvalidator.go]: see 2ab823fe17d72

**Pull request merged**